File: remote/src/main/java/com/kelsos/mbrc/presenters/ArtistAlbumPresenter.java
package com.kelsos.mbrc.presenters;

import com.kelsos.mbrc.annotations.Queue;
import com.kelsos.mbrc.domain.Album;
import com.kelsos.mbrc.ui.views.ArtistAlbumsView;

public interface ArtistAlbumPresenter {
  void load(long artistId);

  void bind(ArtistAlbumsView view);

  void queue(@Queue.Action String action, Album album);
}

File: app/src/main/kotlin/com/kelsos/mbrc/presenters/ArtistAlbumPresenter.kt
package com.kelsos.mbrc.presenters

import com.kelsos.mbrc.annotations.Queue
import com.kelsos.mbrc.domain.Album
import com.kelsos.mbrc.ui.views.ArtistAlbumsView

interface ArtistAlbumPresenter {
  fun load(artistId: Long)

  fun bind(view: ArtistAlbumsView)

  fun queue(@Queue.Action action: String, album: Album)
}
